cmd/controller/app/controller.go

```diff
@@ -55,6 +56,11 @@ func Run(opts *options.ControllerOptions, stopCh <-chan struct{}) {
 
 	run := func(_ <-chan struct{}) {
 		var wg sync.WaitGroup
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			metrics.Default.Start(stopCh)
+		}()
```

Comment:
I guess we only want to run instance of the metrics server then? This `run` function is only called when the controller starts leading.